infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -304,27 +342,29 @@ public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
     /**
      * Resets the log unit server via the BatchWriter.
      * Warning: Clears all data.
-     *
+     * <p>
      * - The epochWaterMark is set to prevent resetting log unit multiple times during
-     *   same epoch.
+     * same epoch.
      * - After this the reset operation is inserted which resets and clears all data.
      * - Finally the cache is invalidated to purge the existing entries.
      */
     @ServerHandler(type = CorfuMsgType.RESET_LOGUNIT)
     private synchronized void resetLogUnit(CorfuPayloadMsg<Long> msg,
                                            ChannelHandlerContext ctx, IServerRouter r) {
+
         // Check if the reset request is with an epoch greater than the last reset epoch seen to
         // prevent multiple reset in the same epoch. and should be equal to the current router
         // epoch to prevent stale reset requests from wiping out the data.
         if (msg.getPayload() > serverContext.getLogUnitEpochWaterMark()
                 && msg.getPayload() == serverContext.getServerEpoch()) {
             serverContext.setLogUnitEpochWaterMark(msg.getPayload());
-            batchWriter.reset(msg.getPayload());
+            batchWriter.addTask(RESET, msg).join();
```

Comment:
Please use the advantages of the non-blocking approach:
```
batchWriter.addTask(RESET, msg).thenRun(() -> {
  dataCache.invalidateAll();
  r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
});
```

Reply:
Done.